packages/react-components/src/components/VideoGallery/TogetherModeStream.tsx

```diff
@@ -0,0 +1,120 @@
+/* @conditional-compile-remove(together-mode) */
+import {
+  ReactionResources,
+  VideoGalleryTogetherModeParticipantPosition,
+  VideoGalleryTogetherModeStreams,
+  TogetherModeStreamViewResult,
+  VideoGalleryLocalParticipant,
+  VideoGalleryRemoteParticipant,
+  VideoStreamOptions
+} from '../../types';
+/* @conditional-compile-remove(together-mode) */
+import { StreamMedia } from '../StreamMedia';
+/* @conditional-compile-remove(together-mode) */
+import { MeetingReactionOverlay } from '../MeetingReactionOverlay';
+/* @conditional-compile-remove(together-mode) */
+import { Stack } from '@fluentui/react';
+/* @conditional-compile-remove(together-mode) */
+import { togetherModeStreamRootStyle } from '../styles/TogetherMode.styles';
+/* @conditional-compile-remove(together-mode) */
+/**
+ * A memoized version of local screen share component. React.memo is used for a performance
+ * boost by memoizing the same rendered component to avoid rerendering this when the parent component rerenders.
+ * https://reactjs.org/docs/react-api.html#reactmemo
+ */
+export const TogetherModeStream = React.memo(
+  (props: {
+    startTogetherModeEnabled?: boolean;
+    isTogetherModeActive?: boolean;
+    onCreateTogetherModeStreamView?: (options?: VideoStreamOptions) => Promise<void | TogetherModeStreamViewResult>;
+    onStartTogetherMode?: (options?: VideoStreamOptions) => Promise<void | TogetherModeStreamViewResult>;
+    onDisposeTogetherModeStreamView?: () => Promise<void>;
+    onSetTogetherModeSceneSize?: (width: number, height: number) => void;
+    togetherModeStreams?: VideoGalleryTogetherModeStreams;
+    seatingCoordinates?: VideoGalleryTogetherModeParticipantPosition;
+    reactionResources?: ReactionResources;
+    localParticipant?: VideoGalleryLocalParticipant;
+    remoteParticipants?: VideoGalleryRemoteParticipant[];
+    screenShareComponent?: JSX.Element;
+    containerWidth?: number;
+    containerHeight?: number;
+  }): React.ReactNode => {
+    const {
+      startTogetherModeEnabled,
+      isTogetherModeActive,
+      onCreateTogetherModeStreamView,
+      onStartTogetherMode,
+      onSetTogetherModeSceneSize,
+      onDisposeTogetherModeStreamView,
+      togetherModeStreams,
+      containerWidth,
+      containerHeight
+    } = props;
+
+    useEffect(() => {
+      return () => {
+        // TODO: Isolate disposing behaviors for screenShare and videoStream
+        onDisposeTogetherModeStreamView && onDisposeTogetherModeStreamView();
+      };
+    }, [onDisposeTogetherModeStreamView]);
+
+    // Trigger startTogetherMode only when needed
+    useEffect(() => {
+      if (startTogetherModeEnabled && !isTogetherModeActive) {
+        onStartTogetherMode?.();
+      }
+    }, [startTogetherModeEnabled, isTogetherModeActive, onStartTogetherMode]);
+
+    // Create stream view if not already created
+    useEffect(() => {
+      if (!togetherModeStreams?.mainVideoStream?.renderElement) {
+        onCreateTogetherModeStreamView?.();
```

Comment:
we should make sure this is defined in the check and then call it. if for some reasion it is not defined in the component then it will have a runtime error.

Reply:
This is the same as onCreateTogetherModeStreamView && onCreateTogetherModeStreamView()